app/frontend/src/search/index.js

```diff
@@ -44,17 +44,19 @@ if (document.querySelector('#vacancies-hits')) {
             return new Promise(resolve => {
                 if (stringMatchesPostcode(query) || (query.length && locations.indexOf(query.toLowerCase()) === -1)) {
                     getGeolocatedCoordinates(query).then(coords => {
-                        coords.success ? enableRadiusSelect() : disableRadiusSelect();
-                        setDataAttribute(document.querySelector('#location'), 'coordinates', `${coords.lat}, ${coords.lng}`);
-                        setDataAttribute(document.querySelector('#radius'), 'radius', document.querySelector('#radius').value || 10);
-                        resolve();
+                        if (coords.success) {
+                            enableRadiusSelect();
+                            setDataAttribute(document.querySelector('#location'), 'coordinates', `${coords.lat}, ${coords.lng}`);
+                            setDataAttribute(document.querySelector('#radius'), 'radius', document.querySelector('#radius').value || 10);
+                            resolve();
+                        }
                     });
-                } else {
```

Comment:
Does the removal of this else statement always disable the radius select input, or does `resolve` on line 51 return from the Promise?

Reply:
yes resolve on 51 stops execution of the promise, returns it in a sense (although you `resolve(something);` if you want a return value so its not quite the same)